src/Exceptionless.AppHost/Extensions/MinIoExtensions.cs

```diff
@@ -0,0 +1,138 @@
+        MinIoBuilder options)
+    {
+        return builder
+            .WithEnvironment("MINIO_ROOT_USER", options.AccessKey ?? "minioadmin")
+            .WithEnvironment("MINIO_ROOT_PASSWORD", options.SecretKey ?? "minioadmin");
+    }
+
+    private static IResourceBuilder<MinIoResource> ConfigureVolume(
+        this IResourceBuilder<MinIoResource> builder,
+        MinIoBuilder options)
+    {
+        if (!string.IsNullOrEmpty(options.DataVolumePath))
+            builder = builder.WithVolume(options.DataVolumePath, "/data");
+
+        return builder;
+    }
+}
+
+public class MinIoResource(string name, string? accessKey = null, string? secretKey = null, string? bucket = "storage")
+    : ContainerResource(name), IResourceWithConnectionString
+{
+    internal const string ApiEndpointName = "api";
+    internal const string ConsoleEndpointName = "console";
+    internal const int DefaultApiPort = 9000;
+    internal const int DefaultConsolePort = 9001;
+
+    private EndpointReference? _apiReference;
+    private EndpointReference? _consoleReference;
+
+    private EndpointReference ApiEndpoint =>
+        _apiReference ??= new EndpointReference(this, ApiEndpointName);
+
+    private EndpointReference ConsoleEndpoint =>
+        _consoleReference ??= new EndpointReference(this, ConsoleEndpointName);
+
+    public ReferenceExpression ConnectionStringExpression =>
+        ReferenceExpression.Create(
+            $"ServiceUrl=http://{ApiEndpoint.Property(EndpointProperty.Host)}:{ApiEndpoint.Property(EndpointProperty.Port)};" +
+            $"AccessKey={AccessKey ?? "minioadmin"};" +
+            $"SecretKey={SecretKey ?? "minioadmin"};" +
+            $"Bucket={Bucket}");
+
+    public string? AccessKey { get; } = accessKey;
+    public string? SecretKey { get; } = secretKey;
+    public string? Bucket { get; } = bucket;
+}
+
+public class MinIoBuilder
+{
+    public int? ApiPort { get; set; }
+    public int? ConsolePort { get; set; }
+    public string? AccessKey { get; set; }
+    public string? SecretKey { get; set; }
+    public string? Bucket { get; set; }
+    public string? DataVolumePath { get; set; }
+
+    public MinIoBuilder WithPorts(int? apiPort = null, int? consolePort = null)
+    {
+        ApiPort = apiPort;
+        ConsolePort = consolePort;
+        return this;
+    }
+
+    public MinIoBuilder WithCredentials(string accessKey, string secretKey)
+    {
+        AccessKey = accessKey;
+        SecretKey = secretKey;
+        return this;
+    }
+
+    public MinIoBuilder WithBucket(string bucket)
+    {
+        Bucket = bucket;
+        return this;
+    }
+
+    public MinIoBuilder WithDataVolume(string path)
+    {
+        DataVolumePath = path;
+        return this;
+    }
+}
+
+internal static class MinIoContainerImageTags
+{
+    internal const string Registry = "docker.io";
+    internal const string Image = "minio/minio";
+    internal const string Tag = "latest";
```

Comment:
would it be better to use known versions as latest will only ever be pulled once.  Makes me wonder if we could pull these versions from an env file?

Reply:
I suppose this should be explicit.